swap/src/cli/cancel_and_refund.rs

```diff
@@ -34,13 +34,12 @@ pub async fn cancel(
         BobState::BtcLocked {
             state3,
             monero_wallet_restore_blockheight,
-            ..
-        } => state3.cancel(monero_wallet_restore_blockheight),
+        } => state3.cancel(monero_wallet_restore_blockheight, None),
```

Comment:
Bob doesn't need the transfer proof in `State6` (CancelTimelockExpired, BtcCancelled, BtcRefunded) because the transfer proof is only need when:
- Redeeming the Monero during the happy path
    - We can never enter back into the happy path after cancelling the swap
- Redeeming the Monero during cooperative redeem
    - During cooperative redeem, Alice will send the transfer proof.